client/app/bundles/course/assessment/submission/components/ScribingView/ScribingCanvas.jsx

```diff
@@ -0,0 +1,549 @@
+        const objects = JSON.parse(scribble.content).objects;
+        const fabricObjs = [];
+
+        // Parse JSON to Fabric.js objects
+        for (let i = 0; i < objects.length; i++) {
+          // eslint-disable-next-line no-undef
+          const klass = fabric.util.getKlass(objects[i].type);
+          switch (objects[i].type) {
+            case 'path': {
+              klass.fromObject(objects[i], (obj) => {
+                this.denormaliseScribble(obj);
+                fabricObjs.push(obj);
+              });
+              break;
+            }
+            case 'i-text':
+            case 'line':
+            case 'rect':
+            case 'ellipse': {
+              const obj = klass.fromObject(objects[i]);
+              this.denormaliseScribble(obj);
+              fabricObjs.push(obj);
+              break;
+            }
+            default: {
+              break;
+            }
+          }
+        }
+
+        // Create layer for each user's scribble
+
+        // Layer for other users' scribble
+        // Disables scribble selection
+        if (isFirstInit && scribble.creator_id !== userId) {
+          // eslint-disable-next-line no-undef
+          const scribbleGroup = new fabric.Group(fabricObjs);
+          scribbleGroup.selectable = false;
+
+          const showLayer = (isShown) => {
+            // eslint-disable-next-line no-param-reassign
+            scribbleGroup._objects.forEach(obj => (obj.visible = isShown));
+          };
+          // Populate layers list
+          const newScribble = {
+            ...scribble,
+            isDisplayed: true,
+            showLayer,
+            scribbleGroup,
+          };
+          this.canvas.layers = [...this.canvas.layers, newScribble];
+          this.canvas.add(scribbleGroup);
+        } else {
+          // Add other user's layers first to avoid blocking of user's layer
+          userScribble = fabricObjs;
+        }
+      });
+
+      // Layer for current user's scribble
+      // Enables scribble selection
+      userScribble.map(obj => (this.canvas.add(obj)));
+    }
+    this.canvas.renderAll();
+    this.isScribblesLoaded = true;
+  }
+
+  initializeCanvas(answerId, imagePath) {
+    const imageUrl = `https://coursemology.org/${imagePath}`;
+    // const imageUrl = `${window.location.origin}/${imagePath}`;
+    this.image = new Image(); // eslint-disable-line no-undef
+    this.image.src = imageUrl;
+
+    this.image.onload = () => {
+      // Get the calculated width of canvas, 750 is min width for scribing toolbar
+      const element = document.getElementById(`canvas-${answerId}`);
+      const maxWidth = Math.max(element.getBoundingClientRect().width, 750);
+
+      this.width = Math.min(this.image.width, maxWidth);
+      this.scale = Math.min(this.width / this.image.width, 1);
+      this.height = this.scale * this.image.height;
+
+      // eslint-disable-next-line no-undef
+      this.canvas = new fabric.Canvas(`canvas-${answerId}`, {
```

Comment:
I tried using the React components before but I can't remember exactly what was the issue faced. The React library lacked documentation (e.g. handling canvas events) so I was primarily using it as a wrapper for Fabric.js instead. Didn't look into their codebase previously to see how I could incorporate them in, but I guess I can look into them now.